### PR TITLE
feat: Atomic function to compute basic stats

### PR DESCRIPTION
**Pull request by naudinlo**
## Fixes

Part of https://github.com/arkhn/dsPrivacy/issues/10

## Description

This PR adds some atomic functions with DP to compute on the client side more complex stats, such as combine variance or student-test. 

## Definition of Done

- [ ] I followed the [Arkhn Code Book](https://www.notion.so/arkhn/Arkhn-Code-Book-23ac18a8af7343d0a3f61f1c9ef7400c) (I swear!).
- [ ] I have written tests for the code I added or updated.
- [ ] I have updated the documentation according to my changes.
- [ ] I have updated the deployment configuration if needed.

### CODE REVIEW

File: R/sumSquaresDP.R [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+#  ' @param x One group to compare with the student-test
+#  ' @param epsilon Privacy budget
+#  ' @param x_min Lower bound for x
+#  ' @param x_max Upper bound for x
+#'
+#' @return a numeric, differentially private sum of squares of the input vector
+#' @export
+
+
+sumSquaresDP <- function(x, epsilon, x_min, x_max){
```
Comment (Jasopaum) [inline R/sumSquaresDP.R:13]: nit: maybe `sumOfSquaresDP` is a bit more precise but 🤷‍♂️

**Review by Jasopaum — APPROVED** [review]
🧹

**Review by Jasopaum — APPROVED** [review]

File: R/numValidDP.R [hunk 2]
```diff
@@ -0,0 +1,14 @@
+#' @title Differential count of valid elements
+#' @description Computes a differentially private count of valid elements of a vector
+#'
+#  ' @param x One group to compare with the student-test
```
Comment (LaRiffle) [inline R/numValidDP.R:4]: You can remove the ref to the `student-test` :)

File: R/sumDP.R [hunk 3]
```diff
@@ -0,0 +1,16 @@
+#' @title Differential sum
+#' @description Computes a differentially private sum of a vector
+#'
+#  ' @param x One group to compare with the student-test
```
Comment (LaRiffle) [inline R/sumDP.R:4]: same

File: R/sumSquaresDP.R [hunk 4]
```diff
@@ -0,0 +1,18 @@
+#' @title Differential sum of squares
+#' @description Computes a differentially private sum of squares of a vector
+#'
+#  ' @param x One group to compare with the student-test
```
Comment (LaRiffle) [inline R/sumSquaresDP.R:4]: same

**Review by LaRiffle — APPROVED** [review]
One remark on the docstring, otherwise it's good to 🚢 !

**Pull request merged**